src/mmdata.h

```diff
@@ -677,6 +849,23 @@ long pntrRevInstr(long start_position, const pntrString *string1,
 flag pntrEq(const pntrString *sout, const pntrString *sin);
 
 /* Add a single null string element to a pntrString - faster than pntrCat */
+/*!
+ * \fn temp_pntrString *pntrAddElement(const pntrString *g)
+ * \param[in] g points to the first element of a NULL terminated array in a
+ *   \ref block.
+ * \return a copy of \p g, the terminal NULL replaced with a \ref vstring ""
+ *   followed by NULL.
+ * \attention   
+ *   - the pointers in \p g are copied to the result.  If some of them
+ *     reference allocated memory, check for possible double free, for example.
+ *   - a pointer to constant data is padded to the right.  The referenced memory
+ *     must not be overwritten.
+ * \post 
+ *   - the elements of \p g are duplicated.
+ *   - a pointer to a NUL byte ("") constant is padded to the right.  Make sure
+ *     the referenced memory is never overwritten.
+ * \bug a pointer to constant data is added as a writeable void*.
```

Comment:
'not incorrect' ...  OK, back to the topic: What makes you sure it is used with vstring only? And not temp_vstring or something else?  If this is a built-in limitation of the function it should be voiced as a precondition, though.  Frog's eye again, I don't know all applications, so cannot decide on this.  But ok, the \post \attention is good enough

Reply:
> What makes you sure it is used with vstring only? And not temp_vstring or something else?

Well, it would be a type error, logically speaking, for it to be used with anything else. It doesn't really make sense to push an unspecified element to a `void*` array unless that element is `NULL`, so since this one isn't it stands to reason that it is only being used for a type which accepts `""` as default value, i.e. `vstring`. `temp_vstring` wouldn't make any sense, you can't hold on to a temp_vstring for any length of time so pushing them to an array is out of the question.

Note that this is reasoning about what the code *should* be doing, and not how it actually uses the function (what you call the "frogs-eye view"). You could also call it "inferred preconditions". I also don't know all the uses of the function, but the typing discipline is strong enough that I can say that with reasonable confidence.

Reply:
Instead of a hidden "inferred precondition" be explicit.  Given your reply I think I should add a
\pre
  Intended to be used with arrays of vstring* only
and remove the attention and matching post condition

Reply:
Yes, I'm talking about inferring what precondition to put in the documentation given that we're both learning what the invariants are on the fly. I think it makes sense to add preconditions like that based on the hypothesis that the code author is being reasonable and revise later if we need to refine it for some unusual situation.